README.md

```diff
@@ -1,52 +1,167 @@
+    $("#sentences").html(response);
+  });
+});
+```
+Using your terminal run the following command from inside the examples folder.
+
+```
+$ python -m SimpleHTTPServer
+```
+
+This starts a simple server that will serve our files over HTTP. Browse to [http://localhost:8000](http://localhost:8000/) and watch as the Ajax request is made and the new data is added to our web page. Pretty cool! We used the power of Ajax to load data from the ```html/sentence.html```. This same idea can be applied to calling the GitHub API or our Rails application.
+
+##Callbacks
+Ajax follows an asynchronous pattern which makes them non-blocking. This means we don't sit around and wait for the request to finish before running the rest of our code. We set callbacks and then fire and forget. When the request is complete, the callbacks are responsible for deciding what to do with the returned data.
+
+If we look at our last example, the Ajax request completed very quick but this won't always be the case. If we request data from a slow server over a slow internet connection, it might take multiple seconds to get a response. Using a callback allows our code to make our request and continue doing other things until the request completes.
+
+To make this concept stick, let's look at a real world example. When you put food into a microwave, you don't stand there and wait for the food to finish cooking. You probably pick up your phone, look at Instagram, read some text messages and of course, work on [Learn](https://learn.co). Basically, you are doing other things while the microwave takes care of cooking your food. When the food is cooked, the microwave beeps and you remove the food and eat it. This final step of removing the food and eating it is exactly how our callbacks work.
+
+## Handling Problems
+So far, we have been dealing with succesful API requests. But things don't always go according to plan.  What happens if the API we are using doesn't respond? Or if we attempt to retrieve a resource that doesn't exist?
+
+This can happen when API requests are based on user input.  Let's go back to the GitHub API where we are retrieving commits.  Imagine we want to retrieve a specific commit using a SHA.
 
-ajax communicates with database without need for page refresh - helps with single page applications
+Postman:
+https://api.github.com/repos/rails/rails/commits?sha=82885325e04d78fb7ec608a4670164d842d23078
+Returns success
 
-use AJAX `get` method to retrieve some text from `index/sentence.html`  and display the text in index.html
+Postman error:
+https://api.github.com/repos/rails/rails/commits?sha=fake-SHA
+Returns a 404 not found
 
-use `python -m SimpleHTTPServer` to serve page to make request
+A good developer will make sure to handle these unexpected events gracefully when using Ajax. We can provide multiple callbacks when using jQuery: one to handle a successful response and one to handle when an error occurs.
 
- 
-## Callbacks
+```javascript
```

Comment:
Can we safely assume the student understands method chaining? This might be another example in favor of plain XHR — you can just add an `if` block (or a `switch` statement), giving the student a sense of what’s really going on under jQuery’s hood without introducing a somewhat confusing concept like method chaining.